soc/xtensa/intel_adsp/Kconfig

```diff
@@ -47,6 +47,20 @@ config ADSP_WIN0_OFFSET
 	  "firmware status register" at boot) within the reserved
 	  region at the start of HP-SRAM.
 
+config ADSP_WIN1_OFFSET
+	int "Byte offset of host window one"
+	default 24576
```

Comment:
FWIW: I hate this scheme, it's wasteful and kludgey; it's really here only because SOF still touches some of this memory via its own offsets (or did at the time) and I was afraid to destabilize things vs. the non-Zephyr firmware builds.  While you're in here, it might be productive to figure out exactly where SOF is getting these values, make 100% sure they're coming only from the platform layer (which means "zephyr") and then kill this kconfig stuff.  Just make the windows regular memory arrays (with appropriate alignment/size restrictions) and then we can reclaim the big header at the front of memory and eliminate the linker code to manage it.

Obviously not a requirement, but a helpful "extra credit" kind of thing, heh.

Reply:
@andyross But these are FW build time settings that we want to be able to configure.  E.g. we now have a bit of a conflict with @nklayman  with my https://github.com/zephyrproject-rtos/zephyr/pull/49147 . We both want to use WIN2, but for different purposes. But this is actually ok and something that can happen. We have now multiple schemes for the host-fw interface, one so called IPC3 and the newer one IPC4. In latter, WIN2 is defined as a variable size window that host needs to discover by looking at the descriptors that are stored at base of WIN2 (I do this in my PR). In IPC3, it's a fixed contract where both host and fw have a shared view semantics and sizes of the slots.

So with that, I think the Kconfig options are actually a reasonable way to configure these.

Reply:
Sorry, for clarity: my complaint is with the linker-reserved fixed-size region, not kconfig per se.  Absolutely let's have a build-time configurable size, but let's use it to size a regular (well, aligned) C array that then get used in a normal way by the linker and can be referenced as a regular symbol in the driver.  Then the linker gets to pack these tightly, and in the future (e.g. when the HP-SRAM bank power management arrives) can more easily place it in "pinned" memory that doesn't get shut off, etc...  (or not.. the logging and IPC windows would want to stay alive, but maybe not the others, etc...).

Reply:
@andyross 
> and then we can reclaim the big header at the front of memory and eliminate the linker code to manage it.

What linker code are you referring to here? Can't find anything in the linker script related to this, I might be looking in the wrong place...

Reply:
There's a big chunk of 64k at the start of HP-SRAM that needs to be offset, c.f. CONFIG_HP_SRAM_RESERVE and all the manually offsetting against it in the memory.h header.  All those regions should be regular C symbols, IMHO.